### PR TITLE
Use debouncer to handle button analysis

### PR DESCRIPTION
**Pull request by dastels**

### CODE REVIEW

**Review by brentru — CHANGES_REQUESTED** [review]
@dastels  The response time to quick double-clicks seems slow and misses. 

Could you add getter/setter methods called `debounce_interval` which increase the debouncer's `interval` property?

File: adafruit_cursorcontrol/cursorcontrol_cursormanager.py [hunk 1]
```diff
@@ -174,3 +161,42 @@ def _check_cursor_movement(self, pressed=None):
                 self._cursor.y -= self._cursor.speed
         else:
             raise AttributeError('Board must have a D-Pad or Joystick for use with CursorManager!')
+
+
+class DebouncedCursorManager(CursorManager):
+    """Simple interaction user interface interaction for Adafruit_CursorControl.
```
Comment (brentru) [inline adafruit_cursorcontrol/cursorcontrol_cursormanager.py:167]: Could you explain the purpose of `DebouncedCursorManager` here? 

This is how you mentioned the `DebouncedCursorManager` works to me: 
> The one with the debouncer would provide the additional press, release, is_pressed functionallity

File: adafruit_cursorcontrol/cursorcontrol_cursormanager.py [hunk 2]
```diff
@@ -174,3 +161,42 @@ def _check_cursor_movement(self, pressed=None):
...
+        self._pressed = 0
+        self._debouncer = Debouncer(lambda: bool(self._pressed & self._pad_btns['btn_a']), interval=debounce_interval)
+
+    @property
+    def is_clicked(self):
+        """Returns True if the cursor button was pressed
+        during previous call to update()
+        """
+        return self._debouncer.rose
+    pressed = is_clicked
```
Comment (brentru) [inline adafruit_cursorcontrol/cursorcontrol_cursormanager.py:186]: indent this

**Review by brentru — CHANGES_REQUESTED** [review]

**Comment by dastels** [comment]
Do we really want/need accessors for the debouncer interval, or is passing it as a constructor argument enough?

**Comment by brentru** [comment]
@dastels  I think it's OK as a kwarg.

**Review by brentru — APPROVED** [review]

**Pull request merged**